### PR TITLE
View test check

### PR DESCRIPTION
**Pull request by AliRKat**
created the branch from the legacy code's bug fix branch, this branch only contains changes in ViewTests.cs

### CODE REVIEW

File: Assets/Countly/Plugins/CountlySDK/Services/ViewCountlyService.cs [hunk 1]
```diff
@@ -854,43 +806,13 @@ public async Task RecordOpenViewAsync(string name, IDictionary<string, object> s
...
-
-                CountlyEventModel currentView = new CountlyEventModel(CountlyEventModel.ViewEvent, segment, 1, null, duration);
-                _ = _eventService.RecordEventAsync(currentView);
-            }
-            await Task.CompletedTask;
+        public async Task RecordCloseViewAsync(string name)
+        {
+            lock (LockObj) {
+                Log.Info("[ViewCountlyService] RecordCloseViewAsync: name = " + name);
+                StopViewWithNameInternal(name, null);
```
Comment (arifBurakDemiray) [inline Assets/Countly/Plugins/CountlySDK/Services/ViewCountlyService.cs:813]: stop view with name is not:
- clipping view names

**Review by arifBurakDemiray — CHANGES_REQUESTED** [review]
I have added some comments that might be a missing functionality.
Because currently it seems like they use missing logic. Or they were extra logic and deleted. If the case is extra logic please ping me again, I will approve the PR

**Comment by AliRKat** [comment]
added the missing logic you mentioned @arifBurakDemiray depending on the review from the others we can keep or remove it. adding obviously seemed as the better option, thanks for giving the heads up

**Comment by AliRKat** [comment]
also merged the bug fix branch in here, we can gather all under one PR if you wish to @turtledreams

**Review by arifBurakDemiray — APPROVED** [review]
Thanks for your work A-Dono 🙇

**Review by peterBrxwn — APPROVED** [review]

**Review by turtledreams — CHANGES_REQUESTED** [review]
Oh I see, consent issue found here. So yea fix as mentioned at the other pr and then we can merge em

**Comment by AliRKat** [comment]
fixed both PRs and merged bug fix in here again, you can merge both or close that one, however you wish @turtledreams

**Review by turtledreams — APPROVED** [review]

**Pull request merged**